src/interface/state_descriptor.hpp

```diff
@@ -245,7 +245,34 @@ class StateDescriptor {
   Dictionary<Dictionary<Metadata>> swarmValueMetadataMap_;
 };
 
-using Packages_t = Dictionary<std::shared_ptr<StateDescriptor>>;
+class Packages_t {
+ public:
+  Packages_t() = default;
+  bool Add(const std::shared_ptr<StateDescriptor> &package) {
+    const auto &name = package->label();
+    if (packages_.count(name) == 0) {
+      packages_[name] = package;
+      return true;
+    }
+    return false;
+  }
+  std::shared_ptr<StateDescriptor> &Get(const std::string &name) {
+    return packages_.at(name);
+  }
+  // TODO(JMM): Should this be provided or deleted?
+  // My preference is to delete it, but it's more API breaking.
+  /*
+  std::shared_ptr<StateDescriptor> &operator[](const std::string &name) {
```

Comment:
On the one hand, I prefer the new method names. On the other hand, I always frown on breaking the API unnecessarily. Of course, we're still pre-1.0, so if there's a better API, we should add it now, so I'd say let's not support operator[].

Remove this commented out section and TODO before merge.

Reply:
Will do. I would like to hear what the Athena-PK people say before I delete it though, since they're stakeholders in the API. @pgrete @forrestglines ?

Reply:
No worries here. Better break the API early than late ;)
I'll do a full review tomorrow.

Reply:
Cool, thanks. I'll delete the commented-out code then. :)